### PR TITLE
ci: update nightly version

### PR DESCRIPTION
**Pull request by carllerche**
It has been a while

### CODE REVIEW

**Comment by carllerche** [comment]
I have no idea why i686-unknown-linux-gnu is overflowing the stack now :thinking:

File: tokio/src/macros/loom.rs [hunk 1]
```diff
@@ -1,11 +1,7 @@
 macro_rules! if_loom {
     ($($t:tt)*) => {{
         #[cfg(loom)]
-        const LOOM: bool = true;
-        #[cfg(not(loom))]
-        const LOOM: bool = false;
-
-        if LOOM {
+        {
```
Comment (carllerche) [inline tokio/src/macros/loom.rs:4]: Fixes CI on `i686-unknown-linux-gnu`

**Comment by carllerche** [comment]
woohoo, CI passes

**Comment by NobodyXu** [comment]
Finally, I had to use unstable options to enable sparse registries in #5691

**Review by taiki-e — APPROVED** [review]

**Review by hawkw — APPROVED** [review]

File: tokio/src/macros/loom.rs (hunk 1 above)
Comment (hawkw) [reply tokio/src/macros/loom.rs:4]: ...why were we doing it this way in the first place, anyway? 🤔

**Review by Darksonn — APPROVED** [review]

**Pull request merged**